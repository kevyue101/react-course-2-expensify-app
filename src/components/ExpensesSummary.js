import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => {
    const expenseWord = props.expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(props.expensesTotal/100).format('$0,0.00');
    return (
    <div>
        <div className="page-header">
            <div className="content-container">
                {
                    props.expensesCount === 0 ? 
                    (
                        <h1>No expenses to add up</h1>
                    ) : 
                    (
                        <h1 className="page-header__title">
                        Viewing <span>{props.expensesCount}</span> <span>{expenseWord}</span> totalling <span>{formattedExpensesTotal}</span>
                        </h1>
                    )
                }
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
        {
  
        }      
    </div>
)};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
