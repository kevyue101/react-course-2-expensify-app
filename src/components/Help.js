import React from 'react';

const HelpPage = () => (
    <div>
        This is from my help comp
    </div>
);

export default HelpPage;