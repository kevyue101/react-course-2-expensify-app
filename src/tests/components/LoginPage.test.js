import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let wrapper, startLogin;

test('test rendering of the login page', () => {
    wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should log in when button is clicked', () => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('button').at(0).simulate('click');
    expect(startLogin).toHaveBeenCalled();
});