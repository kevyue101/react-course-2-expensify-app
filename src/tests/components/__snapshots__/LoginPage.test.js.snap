// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test rendering of the login page 1`] = `
<div>
  <input
    placeholder="Username"
    type="text"
  />
  <input
    placeholder="Password"
    type="text"
  />
  <button>
    Login
  </button>
</div>
`;
